src/parse/parser_public.ts

```diff
@@ -230,11 +230,14 @@ export class Parser {
     /**
      *  @deprecated use the **instance** method with the same name instead
      */
-    static performSelfAnalysis(parserInstance: Parser): void {
-        parserInstance.performSelfAnalysis()
+    static performSelfAnalysis(
+        parserInstance: Parser,
+        serializedGrammar?: ISerializedGast[]
+    ): void {
+        parserInstance.performSelfAnalysis(serializedGrammar)
     }
 
-    public performSelfAnalysis(): void {
+    public performSelfAnalysis(serializedGrammar?: ISerializedGast[]): void {
```

Comment:
use config object pattern, meaning accept a single options argument with multiple sub properties.
This will allow to easily extend it in the future.

Reply:
Will do!